frontend/src/store/reducers/settings.js

```diff
@@ -0,0 +1,64 @@
+import {
+  GET_ACCOUNT_USER_INFO_SUCCESS,
+  GET_ACCOUNT_USER_INFO_FAILURE,
+  GET_ALL_USERS_SUCCESS,
+  GET_ALL_USERS_FAILURE,
+  SUBMIT_EDIT_USER_DETAIL_FORM_SUCCESS,
+  SUBMIT_EDIT_USER_DETAIL_FORM_FAILURE,
+  SUBMIT_EDIT_USER_PASSWORD_FORM_SUCCESS,
+  SUBMIT_EDIT_USER_PASSWORD_FORM_FAILURE,
+} from '../actions/settings'
+import {
+  immutableAppendOrUpdate,
+  normalizeRecordsById,
+} from '../../utils/reducerHelpers'
+
+export default (state = { users: {} }, action) => {
+  switch (action.type) {
+    case GET_ACCOUNT_USER_INFO_FAILURE:
+      return {
+        ...state,
+        error: action.error,
+      }
+    case GET_ACCOUNT_USER_INFO_SUCCESS:
+      return {
+        ...state,
+        users: immutableAppendOrUpdate(state.users, action.user),
+      }
+    case GET_ALL_USERS_SUCCESS:
+      return {
+        ...state,
+        users: normalizeRecordsById(action.users),
+      }
+    case GET_ALL_USERS_FAILURE:
+      // TODO handle error
+      return {
+        ...state,
+        error: action.error,
+      }
+    case SUBMIT_EDIT_USER_DETAIL_FORM_SUCCESS:
```

Comment:
Hi @wy0-0yw  , I just noticed this file.  We don't really need to have a setting reducer. Because every updates from setting page actually update the user data.

So, please move the case statements here (except  GET_ALL_USERS_SUCCESS and GET_ALL_USERS_FAILURE ) to user reducer and then delete this file please.

This will also solve the issue of having to refresh every time we update user detail from setting. 🙇